### PR TITLE
Issue73

### PR DESCRIPTION
**Pull request by minurajeeve**
Added google login feature to the dashboard

### CODE REVIEW

File: src/components/google_login/loginComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  const handleFailure = () => {
+    console.log('login failed');
+  };
+
+  return (
+    <div>
+      <p>Google login</p>
+      <GoogleLogin
+        clientId="626018378389-sfff95dsu0orjgvv7lb21dqtuu6ceai7.apps.googleusercontent.com"
```
Comment (pratikmdhr) [inline src/components/google_login/loginComponent.tsx:24]: you can add this as environment variable so you can use it at other places too like the logout comp..

File: src/components/HomeComponent.tsx [hunk 2]
```diff
@@ -1,14 +1,27 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
+import LoginContext from './themes';
+import LoginProvider from './google_login/loginProvider';
 
+// import LoginComponent from './google_login/loginComponent';
+// import LogoutComponent from './google_login/logoutComponent';
 // eslint-disable-next-line func-names
```
Comment (pratikmdhr) [inline src/components/HomeComponent.tsx:7]: you should delete these

**Review by pratikmdhr — CHANGES_REQUESTED** [review]
Hey @minurajeeve , nice work on the PR.. I see that you have implemented the login component as a standalone component with the login state shared with only it's child components which looks fine. Since you have used context API, i have suggested some changes, by which the login state will be available to the entire app and will be useful later. There are also some general comments related to the naming of the context file, and use of the useEffect hook  that you can look at.

File: src/components/HomeComponent.tsx [hunk 3]
```diff
@@ -1,14 +1,27 @@
...
+
 function HomeComponent({ location }: { location: any }) {
+  const [loginState, setLoginState] = useState(
+    localStorage.getItem('loginstate') === 'true',
+  );
+  const value = useMemo(() => ({ loginState, setLoginState }), [loginState]);
```
[the remark this answers is not in the document]
Comment (minurajeeve) [reply src/components/HomeComponent.tsx:13]: Isn't the Home component the top-level component? If not what is the top-level component?

[the remark this answers is not in the document]
Comment (pratikmdhr) [reply src/components/HomeComponent.tsx:13]: App.tsx is the top level one

**Comment by minurajeeve** [comment]
Made the necessary changes

**Pull request closed without merging**